uuid.go

```diff
@@ -344,6 +345,30 @@ func (u *NullUUID) Scan(src interface{}) error {
 	return u.UUID.Scan(src)
 }
 
+// MarshalJSON marshalls the NullUUID as nil or the nested UUID
+func (u NullUUID) MarshalJSON() ([]byte, error) {
+	if u.Valid == false {
+		return json.Marshal(nil)
+	}
+	return json.Marshal(u.UUID)
+}
+
+// UnmarshalJSON unmarshalls a NullUUID
+func (u *NullUUID) UnmarshalJSON(b []byte) error {
+	var s *UUID
+	if err := json.Unmarshal(b, &s); err != nil {
+		return err
+	}
+
+	if s != nil {
+		u.Valid = true
+		u.UUID = *s
+	} else {
+		u.Valid = false
+	}
+	return nil
+}
```

Comment:
So, you removed ```u.Valid = false``` part. I thought it would be better to explicitly initialize all the fields and not rely on default initialization. I don't mind changing that if you want.

As for the comparison with "null", it injects json logic in the function and that should be the responsibility of the json package. Maybe the contents of the buffer is not ```"null"``` but ```"    null    "``` and that comparison would fail, but the json package will parse it correctly.